test/CMakeLists.txt

```diff
@@ -414,13 +414,37 @@ function(rate_added_test NAME)
     endif()
 endfunction()
 
-foreach(TEST ${TESTS})
-    get_filename_component(BASE_NAME ${TEST} NAME_WE)
-    add_test_executable(test_${BASE_NAME} ${TEST})
+set(EXCLUDE_TESTS)
+if(NOT MIOPEN_ENABLE_SQLITE)
+    list(APPEND EXCLUDE_TESTS test_sqlite)
+endif()
+if(NOT MIOPEN_ENABLE_SQLITE OR (NOT MIOPEN_EMBED_DB STREQUAL ""))
+    list(APPEND EXCLUDE_TESTS test_sqlite_perfdb)
+endif()
```

Comment:
Again we are adding test logic to the CMake file, this should just have been disabled in the test itself using the `#if / @endif`

Reply:
Looking through the code, seems like the above logic existed and was removed, why ? 

@atamazov @averinevg @DrizztDoUrden

Reply:
@JehandadKhan I agree. Overlooked this possibility during review. Should be easy to do.

@averinevg In `add_test_executable`, if
```
if(NOT MIOPEN_ENABLE_SQLITE OR (NOT MIOPEN_EMBED_DB STREQUAL ""))
```
and test name is "test_sqlite_perfdb" then `set_tests_properties(${TEST_NAME} PROPERTIES DISABLED On)`. Much simpler than stuff with lists. Can you please make a PR?

Reply:
@JehandadKhan 
> Looking through the code, seems like the above logic existed and was removed, why ?

To not rebuild the binaries. This is a microscopic step towards making a package of tests.